### PR TITLE
New course: Electron - Electron map and setup only

### PR DESCRIPTION
**Pull request by brianamarie**
## Overview
**TL;DR**

This pull request sets the scene for Electron on-demand courses.  For now, they'd live on services.github.com/on-demand, but in the future, they could live elsewhere, like on Electron's site. 

### Questions
- @github/services-training @lee-dohm Are these the right sections we want to create? 

### Next Steps
- Brief review from @github/services-training 
- After this is merged:
  - Set ship dates for each smaller course
  - Build outlines and content for each smaller section

### Review

To review, spin up locally and navigate to `/on-demand/electron/`.

electron

@github/services-training @lee-dohm What do you think? Does this set the scene for the type of content we want to create?

### CODE REVIEW

**Comment by lee-dohm** [comment]
Is "Converting your app" intended to be a section about converting a pre-existing native application to Electron? If so, yeah, they all sound good to me :+1:

**Comment by brianamarie** [comment]
> Is "Converting your app" intended to be a section about converting a pre-existing native application to Electron? If so, yeah, they all sound good to me 👍

@lee-dohm I was actually a little unclear on what that section should be about, but if the need is for that topic, then yes, it absolutely can be! Thank you for the quick comment and added clarity. 😄

**Comment by lee-dohm** [comment]
I'm not entirely sure what the intention was there either. The thing that I discussed with @crichID (other than section 3 and 4) is the difficulty mismatch between building complex UI components in Electron and environments like XCode or Visual Studio, since Electron doesn't have anything to help one build a UI out of the box if you're not familiar with web dev.

**Comment by crichID** [comment]
> Is "Converting your app" intended to be a section about converting a pre-existing native application to Electron?

@lee-dohm great question! We divide our ships into short, medium, and long term goals. The short term goal is to have a course that covers the packaging and distribution topics you mentioned. I picked these first because, combined with the other two topics (creating and converting an app) they provide a good foundation for the other content we discussed. In terms of timelines, we typically bracket our short term goals at ~10 working days so this phase goes quick :wink:

> The thing that I discussed with @crichID (other than section 3 and 4) is the difficulty mismatch between building complex UI components in Electron and environments like XCode or Visual Studio, since Electron doesn't have anything to help one build a UI out of the box if you're not familiar with web dev.

After we build the foundation, we have this on the roadmap to tackle next. You can see more about the goals [here](https://github.com/github/services-training/issues/493#issuecomment-315832219). Does that make more sense?

As for what we are converting in part 2, we would :heart: to hear your thoughts, but the original idea was to start with a web-app (again, thinking short term ship) such as [this](https://github.com/alysonla/hubber-memory-game) or [this](https://github.com/githubtraining/github-games). Thoughts?

**Comment by lee-dohm** [comment]
@crichID All of that sounds awesome :100: Thanks for explaining :heart:

**Comment by brianamarie** [comment]
As I chew on this, @lee-dohm @crichID it seems that converting an existing native app to Electron is in its own space, away from creating one from scratch, packaging it, and sharing it. Do you think it would be OK to remove that step until a later ship, or is it necessary to the flow of the content?

**Comment by crichID** [comment]
> it seems that converting an existing native app to Electron is in its own space, away from creating one from scratch, packaging it, and sharing it. Do you think it would be OK to remove that step until a later ship, or is it necessary to the flow of the content?

@brianamarie the plan was not to convert a native app to Electron in this ship - it was to take an existing web-app [as mentioned above](https://github.com/github/training-kit/pull/528#issuecomment-316204089) and convert it to an Electron app. So, in my mind, I see the sections as follows:

1. **Starting from Scratch with Electron:** I see this as `git init new-folder` -- we are starting our app from the beginning knowing it will be an electron app.
2. **Converting an Existing Web App:** I see this as doing `git init` inside a web-app I have already created and now want to make in to a native app.
3. **Packaging:** So regardless of the path they selected, this step is relevant.
4. **Sharing:** Same as step 3, this step is always relevant.

The native app piece is a pretty big monster we need to think through because we will need to consider where they are coming from (windows vs macOS), the UI concerns in the original post, etc.

**Comment by brianamarie** [comment]
@crichID Perfect, thank you for the clarification. I'll update the outlines in this PR to reflect those goals. 👍

**Comment by beardofedu** [comment]
What about something that looks like this?

image


Couple other thoughts:
1. Link will still exist on On Demand timeline even if we move it to the Electron site for visibility
1. Once we figure out how to do the split timeline, we apply it to GH 102 and 103 (Desktop | CLI)
1. We should probably get links to the GH Pages courses on the GH Pages site (side convo)

**Comment by brianamarie** [comment]
> What about something that looks like this?

@beardofedu I LOVE this. The only concern I have is that I don't know if converting an existing app will be in this small ship because it's such a bigger bear of a topic. 

I am still unclear on why Intro to Electron and Starting from Scratch are different topics, but once I get that in my head I'll be happy to start the outlines.

**Comment by brianamarie** [comment]
After a chat with @crichID via Skype, we decided to start this small ship with three main parts: 
- Starting from scratch
- Packaging Apps
- Sharing Apps

Things that we can work on in the future include: 
- Converting a web app to an electron app
- Converting a native app to an electron app

For the sake of moving along with the outlines, I'm going to close this PR and open it again into an "electron-course" branch. That branch can serve as our dev branch for electron so we can continue developing and iterating on smaller branches without affecting master until we're ready to launch.
